backend/functions/data_functions.py

```
@@ -243,6 +244,87 @@ def collect_pb_effort_activities(self, activity_data: list, access_token: Option
+        # Iterate through each pb effort activity
+        for activity_id in pb_efforts_ids:
+            # Define activity url
+            activities_url = f"https://www.strava.com/api/v3/activities/{activity_id}/streams"
+
+            params = {
+                "keys": "distance,heartrate,time",
+                "key_by_type": "true"
+            }
+
+            # Execute request
+            data = requests.get(
```

Ruff (0.14.8)

Probable use of `requests` call without timeout

(S113)